### PR TITLE
Redcap error 368

### PR DESCRIPTION
**Pull request by spgarbet**
In doing the redcapError refactor, it became obvious config and api_param were repeated code throughout that could be handled via the dots operator to makeApiCall. There were several instances of inconsistent use of these two. Also, merging with config with rcon$config seems broken as that does not result in a list class. I've left that out, and I'm still investigating it's use.

### CODE REVIEW

**Comment by spgarbet** [comment]
Specific concerns here: https://github.com/vubiostat/redcapAPI/pull/377/files#diff-dfc7e399d34a6ff9c54723f9d6ddc0864da4179604b58f7c4c0adc9f0dbb63b2R173

File: R/redcapConnection.R
Comment (spgarbet) [inline R/redcapConnection.R:1]: This is my problem spot, something is wonky here with how it works with makeApiCall.

**Comment by spgarbet** [comment]
══ Results ═════════════════════════════════════════════════
Duration: 1453.2 s

── Skipped tests (9) ───────────────────────────────────────
• Tests to be developed for 3.0.0 release (9):
  test-151-exportRecords-ArgumentValidation.R:6:5,
  test-151-exportRecords-Functionality.R:6:5,
  test-152-exportRecordsWithDags.R:6:5,
  test-153-exportRecordsWithEvents.R:6:5,
  test-155-exportRecordsWithRepeatingInstruments.R:6:5,
  test-180-exportRecordsOffline-ArgumentValidation.R:6:5,
  test-180-exportRecordsOffline-Functionality.R:6:5,
  test-190-exportReports-ArgumentValidation.R:6:5,
  test-190-exportReports-Functionality.R:6:5

── Failed tests ────────────────────────────────────────────
Failure (test-360-createRedcapProject-ArgumentValidation.R:132:5): Validate config, api_param
`createRedcapProject(rcon, project_title = "Just a title", config = list(1))` threw an error with unexpected message.
Expected match: "'config': Must have names"
Actual message: "1 assertions failed:\n * Variable 'rcon$token': All elements must have exactly 64 characters, but element 1 has 32 chararacters."
Backtrace:
    ▆
 1. ├─testthat::expect_error(...) at test-360-createRedcapProject-ArgumentValidation.R:132:5
 2. │ └─testthat:::quasi_capture(...)
 3. │   ├─testthat (local) .capture(...)
 4. │   │ └─base::withCallingHandlers(...)
 5. │   └─rlang::eval_bare(quo_get_expr(.quo), quo_get_env(.quo))
 6. ├─redcapAPI::createRedcapProject(...)
 7. └─redcapAPI:::createRedcapProject.redcapApiConnection(...) at redcapAPI/R/createRedcapProject.R:109:3
 8.   └─checkmate::reportAssertions(coll) at redcapAPI/R/createRedcapProject.R:176:3

Failure (test-360-createRedcapProject-ArgumentValidation.R:136:5): Validate config, api_param
`createRedcapProject(rcon, project_title = "Just a title", config = "not a list")` threw an error with unexpected message.
Expected match: "'config': Must be of type 'list'"
Actual message: "1 assertions failed:\n * Variable 'rcon$token': All elements must have exactly 64 characters, but element 1 has 32 chararacters."
Backtrace:
    ▆
 1. ├─testthat::expect_error(...) at test-360-createRedcapProject-ArgumentValidation.R:136:5
 2. │ └─testthat:::quasi_capture(...)
 3. │   ├─testthat (local) .capture(...)
 4. │   │ └─base::withCallingHandlers(...)
 5. │   └─rlang::eval_bare(quo_get_expr(.quo), quo_get_env(.quo))
 6. ├─redcapAPI::createRedcapProject(...)
 7. └─redcapAPI:::createRedcapProject.redcapApiConnection(...) at redcapAPI/R/createRedcapProject.R:109:3
 8.   └─checkmate::reportAssertions(coll) at redcapAPI/R/createRedcapProject.R:176:3

Failure (test-360-createRedcapProject-ArgumentValidation.R:141:5): Validate config, api_param
`createRedcapProject(rcon, project_title = "Just a title", api_param = list(1))` threw an error with unexpected message.
Expected match: "'api_param': Must have names"
Actual message: "1 assertions failed:\n * Variable 'rcon$token': All elements must have exactly 64 characters, but element 1 has 32 chararacters."
Backtrace:
    ▆
 1. ├─testthat::expect_error(...) at test-360-createRedcapProject-ArgumentValidation.R:141:5
 2. │ └─testthat:::quasi_capture(...)
 3. │   ├─testthat (local) .capture(...)
 4. │   │ └─base::withCallingHandlers(...)
 5. │   └─rlang::eval_bare(quo_get_expr(.quo), quo_get_env(.quo))
 6. ├─redcapAPI::createRedcapProject(...)
 7. └─redcapAPI:::createRedcapProject.redcapApiConnection(...) at redcapAPI/R/createRedcapProject.R:109:3
 8.   └─checkmate::reportAssertions(coll) at redcapAPI/R/createRedcapProject.R:176:3

Failure (test-360-createRedcapProject-ArgumentValidation.R:145:5): Validate config, api_param
`createRedcapProject(rcon, project_title = "Just a title", api_param = "not a list")` threw an error with unexpected message.
Expected match: "'api_param': Must be of type 'list'"
Actual message: "1 assertions failed:\n * Variable 'rcon$token': All elements must have exactly 64 characters, but element 1 has 32 chararacters."
Backtrace:
    ▆
 1. ├─testthat::expect_error(...) at test-360-createRedcapProject-ArgumentValidation.R:145:5
 2. │ └─testthat:::quasi_capture(...)
 3. │   ├─testthat (local) .capture(...)
 4. │   │ └─base::withCallingHandlers(...)
 5. │   └─rlang::eval_bare(quo_get_expr(.quo), quo_get_env(.quo))
 6. ├─redcapAPI::createRedcapProject(...)
 7. └─redcapAPI:::createRedcapProject.redcapApiConnection(...) at redcapAPI/R/createRedcapProject.R:109:3
 8.   └─checkmate::reportAssertions(coll) at redcapAPI/R/createRedcapProject.R:176:3

[ FAIL 4 | WARN 0 | SKIP 9 | PASS 2046 ]

**Comment by spgarbet** [comment]
── R CMD check results ──────────────── redcapAPI 2.9.2 ────
Duration: 31.4s

0 errors ✔ | 0 warnings ✔ | 0 notes ✔

R CMD check succeeded

**Comment by spgarbet** [comment]
```
══ Results ═════════════════════════════════════════════════
Duration: 971.0 s

── Skipped tests (12) ──────────────────────────────────────
• No value provided for EXPORT_REPORTS_ID. Tests Skipped
  (3): test-240-exportTypedReports-Functionality.R:8:5,
  test-240-exportTypedReports-Functionality.R:54:5,
  test-240-exportTypedReports-Functionality.R:72:5
• Tests to be developed for 3.0.0 release (9):
  test-151-exportRecords-ArgumentValidation.R:6:5,
  test-151-exportRecords-Functionality.R:6:5,
  test-152-exportRecordsWithDags.R:6:5,
  test-153-exportRecordsWithEvents.R:6:5,
  test-155-exportRecordsWithRepeatingInstruments.R:6:5,
  test-180-exportRecordsOffline-ArgumentValidation.R:6:5,
  test-180-exportRecordsOffline-Functionality.R:6:5,
  test-190-exportReports-ArgumentValidation.R:6:5,
  test-190-exportReports-Functionality.R:6:5

[ FAIL 0 | WARN 0 | SKIP 12 | PASS 2049 ]

```

**Comment by obregos** [comment]
[ FAIL 0 | WARN 0 | SKIP 12 | PASS 2049 ]

**Review by obregos — APPROVED** [review]
[ FAIL 0 | WARN 0 | SKIP 12 | PASS 2049 ]

**Pull request merged**